src/printer.rs

```diff
@@ -137,11 +137,20 @@ impl<'a, 'b, 'c, W: std::fmt::Write + Sized> Printer<'a, 'b, 'c, W> {
   }
 
   /// Writes a raw string to the underlying destination.
-  ///
-  /// NOTE: Is is assumed that the string does not contain any newline characters.
-  /// If such a string is written, it will break source maps.
   pub fn write_str(&mut self, s: &str) -> Result<(), PrinterError> {
-    self.col += s.len() as u32;
+    let mut last_line_start: usize = 0;
+
+    for (idx, n) in s.char_indices() {
```

Comment:
Maybe we could move this to a separate function, like `write_str_with_newlines`. `write_str` is very hot since it is called so much, and most of the time there are no newlines. Moving this to a separate function would keep it fast in the common case.